blue_only.py

```diff
@@ -1,221 +1,290 @@
+                config = yaml.safe_load(file)
+            except yaml.YAMLError as exc:
+                print(f"Error parsing YAML file: {exc}")
+                return -1
+    except FileNotFoundError:
+        print(f"File not found: {CONFIG_FILE_PATH}")
+        return -1
+    except IOError as exc:
+        print(f"Error when opening file: {exc}")
+
+    parser = argparse.ArgumentParser(description="AEAC 2024 Auto-landing script")
+    parser.add_argument(
+        "--method",
+        help="Method for selecting landing pads",
+        choices=["contour", "yolo"],
+        default="contour",
+    )
+    args = parser.parse_args()
+
+    dotenv.load_dotenv(".key")
+    secret_key = os.getenv("KEY")
+
+    cam = cv2.VideoCapture(CAMERA)
+
+    vehicle = pymavlink.mavutil.mavlink_connection("tcp:localhost:14550")
+    # vehicle = pymavlink.mavutil.mavlink_connection('/dev/ttyUSB0', baud=57600)
+    vehicle.wait_heartbeat()
+    print(
+        f"Heartbeat from system (system {vehicle.target_system} component {vehicle.target_component})"
+    )
+
+    pos_message = vehicle.mav.command_long_encode(
+        vehicle.target_system,  # Target system ID
+        vehicle.target_component,  # Target component ID
+        pymavlink.mavutil.mavlink.MAV_CMD_SET_MESSAGE_INTERVAL,  # ID of command to send
+        0,  # Confirmation
+        33,  # param1: Message ID to be streamed
+        250000,  # param2: Interval in microseconds
+        0,  # param3 (unused)
+        0,  # param4 (unused)
+        0,  # param5 (unused)
+        0,  # param5 (unused)
+        0,  # param6 (unused)
+    )
+
+    secret_key = bytearray(secret_key, "utf-8")
+    vehicle.setup_signing(secret_key, True, my_allow_unsigned_callback, int(time.time()), 0)
+    vehicle.mav.send(pos_message)
+
+    altitude_m = 0
+    loop_counter = 0
+    last_time = current_milli_time()
+    last_image_time = current_milli_time()
+
+    while True:
+        is_pad_detected = False
+        result, image = cam.read()
+        if not result:
+            print("ERROR: Could not get image from camera")
+            continue
+
+        image = cv2.flip(image, 0)
+        image = cv2.flip(image, 1)
+        im_h, im_w, _ = image.shape
+        print("Input image width: " + str(im_w))
+        print("Input image height: " + str(im_h))
+        try:
+            altitude_mm = vehicle.messages[
+                "GLOBAL_POSITION_INT"
+            ].relative_alt  # Note, you can access message fields as attributes!
+            altitude_m = max(altitude_mm / 1000, 0.0)
+            print("Altitude AGL: ", altitude_m)
+        except (KeyError, AttributeError):
+            print("No GLOBAL_POSITION_INT message received")
+            continue
+
+        if args.method == "contour":
+            bounding_box = detect_landing_pads_contour(image)
+        elif args.method == "yolo":
+            bounding_box = detect_landing_pads_yolo(image, config)
+
+        if bounding_box is not None:
```

Comment:
from above, let's do `result, bounding_box = detect(image)` and check the result instead of the bounding box to see if it's valid